pkg/resources/allocator.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 Intel Corp. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"sort"
+
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+)
+
+// DeviceSet is to hold and manipulate a set of HostDevice
+type DeviceSet map[string]types.HostDevice
+
+// PackedAllocator implements the Allocator interface
+type PackedAllocator struct {
+}
+
+// Allocator implements the Allocator interface
+type Allocator struct {
+}
+
+// NewPackedAllocator create instance of PackedAllocator
+func NewPackedAllocator() types.Allocator {
+	return &PackedAllocator{}
+}
+
+// NewAllocator create instance of Allocator
+func NewAllocator() types.Allocator {
+	return &Allocator{}
+}
+
+// NewDeviceSet is to create an empty DeviceSet
+func NewDeviceSet() DeviceSet {
+	set := make(DeviceSet)
+	return set
+}
+
+// Insert is to add a HostDevice in DeviceSet
+func (ds DeviceSet) Insert(pciAddr string, device types.HostDevice) {
+	ds[pciAddr] = device
+}
+
+// Delete is to delete a HostDevice in DeviceSet
+func (ds DeviceSet) Delete(pciAddr string) {
+	delete(ds, pciAddr)
+}
+
+// Sort is to sort the DeviceSet and return the sorted keys
+func (ds DeviceSet) Sort() []string {
```

Comment:
this reads like you can sort the DeviceSet. IMO its a bit confusing

how about renaming this method to : AsSortedStrings/AsSorted/AsSortedSlice ?